jsonnet/kube-state-metrics.libsonnet

```diff
@@ -110,6 +110,7 @@ function(params)
                     c {
                       args+: [
                         '--metric-denylist=kube_secret_labels',
+                        '--metric-labels-allowlist=pods=[*]',
```

Comment:
it doesn't work (yet) because it needs https://github.com/kubernetes/kube-state-metrics/commit/ba36af0daecc2979657ccec784c0a1e600b9d1e1 which hasn't been released yet. 

When we bumped our downstream kube-state-metrics fork, I introduced a temporary patch to collect all labels for node resources: https://github.com/openshift/kube-state-metrics/commit/10c3d722d8c3010596a7cae64a006d9ff7aaa4a3. I did it this way because the upstream patch adding wildcard support wasn't merged yet and the origin e2e tests wouldn't pass without it.

It might be worth doing the following:
1) cherry-pick the upstream patch in origin/kube-state-metrics
2) pass `--metric-labels-allowlist=pods=[*],node=[*]`
3) remove the carry-patch in origin/kube-state-metrics